apps/docs/app/utils/menu.server.ts

```diff
@@ -0,0 +1,39 @@
+import { getClient } from "./sanity/client";
+
+export type MenuItem = {
+  _type: "menuItem" | "divider";
+  title: string;
+  tags: string[];
+  url: string;
+  subItems?: MenuItem[];
+};
+export type Menu = {
+  slug: string;
+  menuItems: MenuItem[];
+};
+export const getMenus = async () =>
+  getClient().fetch<MenuItem[]>(
+    `*[_type == "menu"] { 
+      "slug": slug.current,
+      "menuItems": menuItems[]{
+        _type,
+        title,
+        "url": select(
+          defined(internalLink) => 
+            "/" + internalLink->category->slug.current + 
+            "/" + internalLink->slug.current, 
+          defined(externalLink) => externalLink
+        ),
+        subItems[]{
+          title,
+          tags,
+          "url": select(
+            defined(internalLink) => 
+              "/" + internalLink->category->slug.current + 
+              "/" + internalLink->slug.current, 
+            defined(externalLink) => externalLink
+          ),
+        }
+      } 
+    }`
+  );
```

Comment:
Dette er den mest avanserte spørringen jeg noensinne har skrevet i GROQ, men det den gjør er ganske straight forward.

Den henter alle menyer, og henter ut menu items. For hver menu item returnerer jeg et felt "url", som – om man har definert en internalLink (en artikkel) – setter sammen URLen basert på kategori og slug, og ellers bare returnerer hva enn man skrev inn i externalLink feltet.

Også gjør den det samme for subItems.

`select` var en ganske kul funksjon!